### PR TITLE
Replaces yargs with commander.

### PR DESCRIPTION
**Pull request by MicahZoltu**
Works towards resolving #361.  Yargs brings in ~26 packages while commander brings in 1.

### CODE REVIEW

**Comment by MicahZoltu** [comment]
It is worth noting, I'm leaning on existing tests covering this.  I did do a run of `node ./solcjs --bin test/contracts/Smoke.sol` and it appeared to "do the right thing".  I also tried with `--abi` which also appears to have worked.

**Comment by MicahZoltu** [comment]
Looking into test failures now (the tests won't run on my machine for some reason).

File: solcjs [hunk 1]
```diff
@@ -11,48 +11,29 @@ var smtsolver = require('./smtsolver.js');
...
+
+var program = new commander.Command();
+program.name('solcjs');
+program.version(solc.version());
+program
+  .option('--version', 'Show version and exit.')
+  .option('--optimize', 'Enable bytecode optimizer.')
+  .option('--bin', 'Binary of the contracts in hex.')
+  .option('--abi', 'ABI of the contracts.')
+  .option('--standard-json', 'Standard JSON configuration file.')
```
Comment (axic) [inline solcjs]: This doesn't take a file name so that the help text is confusing. I'd suggest to stick to the original text.

Comment (MicahZoltu) [reply solcjs]: Oops, I have no idea why I changed that.  Fixed.

Comment (MicahZoltu) [reply solcjs]: Ugh, I pushed the wrong branch... give me a few minutes while I clean up the mess I just created for myself.

Comment (MicahZoltu) [reply solcjs]: Fixed.

**Pull request merged**